lib/AnsiSerializer.js

```diff
@@ -127,7 +128,7 @@ AnsiSerializer.prototype.text = function (options) {
 
                 var open = ansiStyles[styleName].open;
                 var close = ansiStyles[styleName].close;
-                if (color16Hex !== color256Hex) {
+                if (this.supportsColor.has256) {
```

Comment:
Mhh we are also using the 256 colors in the case where the hex is the same as the 16 color as this test was removed: `color16Hex !== color256Hex`.